src-electron/generator/helper-attribute.js

```diff
@@ -65,6 +66,77 @@ async function attributeDefault(options) {
   return templateUtil.templatePromise(this.global, p)
 }
 
+/**
+ * If helper that checks if an attribute is not supported.
+ * An attribute is not supported when it is a struct. However it is supported
+ * if it is an array of structs.
+ * @param {*} type
+ * @param {*} isArray
+ * @param {*} options
+ * @returns Promise of content
+ */
+async function if_unsupported_attribute_type(type, isArray, options) {
+  let struct = null
+  if (isArray) {
+    return options.inverse(this)
+  } else {
+    let packageIds = await templateUtil.ensureZclPackageIds(this)
+    if (type && typeof type == 'string') {
+      struct = await queryZcl.selectStructByName(
```

Comment:
Won't this fail if structs with the same name are defined in multiple clusters?

Reply:
Fixed this to include clusterIds as well